pkg/workerpool/observability.go

```diff
@@ -27,75 +27,117 @@ const (
 )
 
 type workPoolMetrics struct {
-	workersStarted *metrics.ContextualCounterVec
-	workersStopped *metrics.ContextualCounterVec
-	workEnqueued   *metrics.ContextualCounterVec
-	workDequeued   *metrics.ContextualCounterVec
+	workersStarted *prometheus.CounterVec
+	workersIdle    *prometheus.GaugeVec
+	workersStopped *prometheus.CounterVec
+	workQueueSize  *prometheus.GaugeVec
+	workProcessed  *metrics.ContextualCounterVec
+	workDropped    *metrics.ContextualCounterVec
```

Comment:
Is this to filter on the namespace? What's in the context that's needed for these 2 vectors alone?